### ISSUE TITLE
Feature: sublabels / option slots

### ISSUE THREAD

**Issue by ernestoalejo** [riophae/vue-treeselect#8]
I would like to customize the options adding a new gray sublabel, like count but with my own text. The best solution would probably be a slot that allows the users of this component to change the labels with our own custom HTML:

```html
<treeselect ...>
  <template slot="option" slot-scope="option">
    {{option.label}}
    ...other things i want to customize the label with here...
  </template>
</treeselect>
```

You can provide a default slot implementation as a fallback, so it's a completely backwards compatible change and only adds new functionality.

**Comment by riophae** [comment]
I totally agree. Thanks for your feedback!

**Comment by riophae** [comment]
Hey, I just added the slot. It would be like this to use it:

```html
<treeselect ...>
  <template slot="option" slot-scope="{ option }">
    {{option.label}}
    ...other things i want to customize the label with here...
  </template>
</treeselect>
```
Note that, `slot-scope="{ option }"` used [destructuring](https://vuejs.org/v2/guide/components.html#Destructuring). The `option` variable provides a lot of properties (e.g. `isBranch`, `isLeaf`, `ancestors`, `level`, ...etc) for your template to consume. You can also use `option.raw` to access the original option data you passed to the component.

Do you think the API design fits your needs?

**Comment by anfinil** [comment]
Any chance to release a new version?

**Comment by riophae** [comment]
@anfinil I'm sorry, it's not done yet :(

**Comment by riophae** [comment]
Hi, really sorry it took too long to have this feature finished. Just to let you know I just shipped it in v0.0.7.
